assets/source/setup-guide/app/components/Billing/Billing.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { __ } from '@wordpress/i18n';
+import { Spinner } from '@woocommerce/components';
+import { recordEvent } from '@woocommerce/tracks';
+import {
+	Button,
+	CardBody,
+	Flex,
+	FlexItem,
+	FlexBlock,
+	__experimentalText as Text, // eslint-disable-line @wordpress/no-unsafe-wp-apis --- _experimentalText unlikely to change/disappear and also used by WC Core
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+/**
+ * Clicking on "Setup Billing" button.
+ *
+ * @event wcadmin_pfw_billing_setup_button_click
+ */
+
+/**
+ * Clicking on "Go to billing settings" button.
+ *
+ * @event wcadmin_pfw_go_to_billing_button_click
+ */
+
+const Billing = ( { accountData } ) => {
+	const isBillingSetup = accountData?.is_billing_setup;
+	const statusLabe = classnames( 'pfw-billing-info', {
+		'pfw-billing-info--status-success': isBillingSetup === true,
+		'pfw-billing-info--status-error': isBillingSetup === false,
+	} );
+
+	let element = null;
+
+	if ( isBillingSetup === undefined ) {
+		element = <Spinner className="billing-info__preloader" />;
+	}
+
+	if ( isBillingSetup === true ) {
+		element = (
+			<>
+				<FlexBlock className={ statusLabe }>
+					<Text variant="body">
+						{ __(
+							'Billing Setup Correctly',
+							'pinterest-for-woocommerce'
+						) }
+					</Text>
+				</FlexBlock>
+
+				<FlexItem>
+					<Button
+						variant="link"
+						href={
+							wcSettings.pinterest_for_woocommerce
+								.billingSettingsUrl
+						}
+						onClick={ () =>
+							recordEvent( 'pfw_go_to_billing_button_click' )
+						}
+					>
+						{ __(
+							'Go to billing settings',
+							'pinterest-for-woocommerce'
+						) }
+					</Button>
+				</FlexItem>
+			</>
+		);
+	}
+
+	if ( isBillingSetup === false ) {
+		element = (
+			<>
+				<FlexBlock className={ statusLabe }>
+					<Text variant="body">
+						{ __(
+							'No Valid Billing Setup Found',
+							'pinterest-for-woocommerce'
+						) }
+					</Text>
+				</FlexBlock>
+
+				<FlexItem>
+					<Button
+						variant="primary"
```

Comment:
With this code, it would produce this UI:

<img width="460" alt="Screenshot 2024-04-18 at 4 19 41 PM" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/417342/d332ac8e-4e07-4992-b20d-1ea270493765">

I see two issues with this:

- The button text is black and it is not obvious that it is a button.
- It has a bit more padding on the right and it is not aligned with the "Disconnect" button above it.

We should use `isLink` without "Primary".

```suggestion
						isLink
```

It would produce the following UI:

<img width="452" alt="image" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/417342/96759de3-ecd2-4666-b9c3-dc76c73ef77d">

Reply:
With isLink I see:
<img width="450" alt="image" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/17271089/115752cf-f786-427e-a9c6-e60ed9871cbf">

Do you maybe know why?

Reply:
I'm seeing the same thing. I think it is just TypeScript / code editor resolving to wrong type definition files, i.e. we are using and bundling old version of @wordpress/components, but it is resolved to newer version of definition files.